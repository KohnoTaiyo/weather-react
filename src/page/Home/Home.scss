@use "../../styles/variables" as *;

.current {
  display: flex;
  align-items: center;
  column-gap: 8px;
  background-color: $primaryColor;
  color: $whiteColor;
  padding: 8px;
  border-radius: 4px;
  margin-bottom: 16px;

  &__name {
    font-size: $largeFontSize;
    font-weight: bold;
  }

  &__temp {
    margin-right: 4px;

    &Main {
      font-size: $titleFontSize;
    }
  }
}

.forecast {
  display: flex;
  justify-content: space-between;
  background-color: $primaryColor;
  color: $whiteColor;
  border-radius: 4px;

  &__item {
    width: 100%;
    text-align: center;

    &:not(:last-child) {
      border-right: 1px solid $whiteColor;
    }

    &Date {
      border-bottom: 1px solid $whiteColor;
      padding: 8px;
      font-weight: bold;
    }

    &Name {
      padding: 0 8px 16px;
    }
  }
}
