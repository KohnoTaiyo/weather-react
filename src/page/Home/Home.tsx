import "./Home.scss";

import { Link } from "react-router-dom";

import { formatDate } from "../../hooks/formatDate";
import { ForecastFewDaysWeatherType, StateParams } from "../../types";

function Home({ forecastWeather }: { forecastWeather: ForecastFewDaysWeatherType }) {
  const { current, forecast, location } = forecastWeather;
  const locationName = [location?.name, location?.region, location?.country].filter(Boolean).join(", ");

  return (
    <>
      <h3>{`Weather in ${locationName} at ${location.localtime}`}</h3>
      <div className="current">
        <img
          src={`https://${current.condition.icon}`}
          className="current_icon"
          alt={`${current.condition.text} Image`}
        />
        <p className="current__name">{current.condition.text}</p>
        <p className="current__temp">
          <span className="current__tempMain">{current.temp_c}</span>Â°C
        </p>
        <p>{`Humidity: ${current.humidity}%`}</p>
        <p>{`Wind: ${current.wind_kph}kph`}</p>
      </div>
      <h3>5days weather</h3>
      <ul className="forecast">
        {forecast.forecastday.map((item) => {
          const sendPathParams: StateParams = { location: location?.name, date: item.date };
          return (
            <Link
              to={{ pathname: `/${location?.name.replace(/\s+/g, "")}` }}
              state={sendPathParams}
              key={item.date}
              className="forecast__item"
            >
              <li>
                <p className="forecast__itemDate">{formatDate(item.date)}</p>
                <img src={item.day.condition.icon} alt="" />
                <p className="forecast__itemName">{item.day.condition.text}</p>
              </li>
            </Link>
          );
        })}
      </ul>
    </>
  );
}

export default Home;
