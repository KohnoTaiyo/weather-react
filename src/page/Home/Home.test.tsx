import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import { mockForecastWeather } from "../../mock";
import Home from "./Home";

describe("Home", () => {
  test("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <Home forecastWeather={mockForecastWeather} />
      </BrowserRouter>,
    );
    expect(screen.getByText(/Weather in New York, New York, United States of America at/i)).toBeInTheDocument();
    expect(screen.getByText(/5days weather/i)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
