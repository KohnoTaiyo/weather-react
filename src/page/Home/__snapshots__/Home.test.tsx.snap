// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <h3>
    Weather in New York, New York, United States of America at 2023-12-08 9:02
  </h3>
  <div
    class="current"
  >
    <img
      alt="Sunny Image"
      class="current_icon"
      src="https:////cdn.weatherapi.com/weather/64x64/day/113.png"
    />
    <p
      class="current__name"
    >
      Sunny
    </p>
    <p
      class="current__temp"
    >
      <span
        class="current__tempMain"
      >
        5
      </span>
      Â°C
    </p>
    <p>
      Humidity: 65%
    </p>
    <p>
      Wind: 3.6kph
    </p>
  </div>
  <h3>
    5days weather
  </h3>
  <ul
    class="forecast"
  >
    <a
      class="forecast__item"
      href="/NewYork"
    >
      <li>
        <p
          class="forecast__itemDate"
        >
          12/08
        </p>
        <img
          alt=""
          src="//cdn.weatherapi.com/weather/64x64/day/143.png"
        />
        <p
          class="forecast__itemName"
        >
          Mist
        </p>
      </li>
    </a>
    <a
      class="forecast__item"
      href="/NewYork"
    >
      <li>
        <p
          class="forecast__itemDate"
        >
          12/09
        </p>
        <img
          alt=""
          src="//cdn.weatherapi.com/weather/64x64/day/119.png"
        />
        <p
          class="forecast__itemName"
        >
          Cloudy
        </p>
      </li>
    </a>
    <a
      class="forecast__item"
      href="/NewYork"
    >
      <li>
        <p
          class="forecast__itemDate"
        >
          12/10
        </p>
        <img
          alt=""
          src="//cdn.weatherapi.com/weather/64x64/day/308.png"
        />
        <p
          class="forecast__itemName"
        >
          Heavy rain
        </p>
      </li>
    </a>
    <a
      class="forecast__item"
      href="/NewYork"
    >
      <li>
        <p
          class="forecast__itemDate"
        >
          12/11
        </p>
        <img
          alt=""
          src="//cdn.weatherapi.com/weather/64x64/day/308.png"
        />
        <p
          class="forecast__itemName"
        >
          Heavy rain
        </p>
      </li>
    </a>
    <a
      class="forecast__item"
      href="/NewYork"
    >
      <li>
        <p
          class="forecast__itemDate"
        >
          12/12
        </p>
        <img
          alt=""
          src="//cdn.weatherapi.com/weather/64x64/day/113.png"
        />
        <p
          class="forecast__itemName"
        >
          Sunny
        </p>
      </li>
    </a>
  </ul>
</div>
`;
